x/concentrated-liquidity/internal/swapstrategy/one_for_zero.go

```diff
@@ -59,7 +59,9 @@ func (s oneForZeroStrategy) ComputeSwapStep(sqrtPriceCurrent, sqrtPriceNextTick,
 		sqrtPriceNext = sqrtPriceTarget
 	} else {
 		// Otherwise, compute the next sqrt price based on the amount remaining after fee.
-		sqrtPriceNext = math.GetNextSqrtPriceFromAmount1RoundingDown(sqrtPriceCurrent, liquidity, amountRemainingLessFee)
+		// TODO: when swapping in given out, GetNextSqrtPriceFromAmount1OutRoundingDown must be used.
```

Comment:
This might be a discussion for later on, but I suppose we could try using boolean fields for in / out as we do for zero / one

Reply:
I ended up defining a seprate API:
- `ComputeSwapStepOutGivenIn` and `ComputeSwapStepInGivenOut`

Working on that for #4427.

I think it made it much simpler and clear in terms of the differences between the methods. But yeah let's discuss this further in the review once ready

Reply:
This direction also sgtm